runtime/src/lib.rs

```diff
@@ -1147,121 +1150,6 @@ pub type SignedExtra = (
     pallet_commitments::CommitmentsSignedExtension<Runtime>,
 );
 
-mod account_data_migration {
-    use super::*;
-    use frame_support::log;
-    use pallet_balances::ExtraFlags;
-
-    mod prev {
-        use super::*;
-        use frame_support::{pallet_prelude::ValueQuery, storage_alias, Blake2_128Concat};
-
-        #[derive(Encode, Decode, Clone, PartialEq, Eq, Default, Debug)]
-        pub struct AccountData<Balance> {
-            pub free: Balance,
-            pub reserved: Balance,
-            pub misc_frozen: Balance,
-            pub fee_frozen: Balance,
-        }
-
-        #[storage_alias]
-        pub type Account<T: frame_system::pallet::Config> = StorageMap<
-            frame_system::pallet::Pallet<T>,
-            Blake2_128Concat,
-            AccountId,
-            AccountData<Balance>,
-            ValueQuery,
-        >;
-    }
-
-    const TARGET: &'static str = "runtime::account_data_migration";
-    pub struct Migration;
-    impl OnRuntimeUpgrade for Migration {
-        /// Save pre-upgrade account ids to check are decodable post-upgrade.
-        #[cfg(feature = "try-runtime")]
-        fn pre_upgrade() -> Result<Vec<u8>, sp_runtime::TryRuntimeError> {
-            let account_ids = prev::Account::<Runtime>::iter_keys().collect::<Vec<_>>();
-            log::info!(target: TARGET, "pre-upgrade");
-
-            Ok(account_ids.encode())
-        }
-
-        /// Ensures post-upgrade that
-        /// 1. Number of accounts has not changed.
-        /// 2. Each account exists in storage and decodes.
-        /// 3. Each account has a provider val >0.
```

Comment:
Was this something you wrote? I was thinking how it was even possible to get an old lifecycle runtime upgrade hook from upstream to execute in our chain, but if you wrote a brand new one, it does the job just fine.

Reply:
Yeah I wrote this from scratch.